7a_temp_coop_munge/src/data_parsers/parse_mo_usace.R

```diff
@@ -0,0 +1,427 @@
+    backfill_site_ids(.) %>%
+
+    # Removes all partially populated row created by the
+    # previous two munge steps
+    remove_incomplete_rows(.) %>%
+    convert_date(.) %>%
+    dplyr::bind_rows(.)
+
+}
+
+parse_mo_usace_2010 <- function(filepath, keep_cols,
+                                subset_sheets = c('PT', 'RA'),
+                                subset_skip = 1, remove_sheets) {
+
+  # list all sheets in the df and read each one
+  dat <- read_usace_sheets(filepath, subset_sheets, subset_skip) %>%
+    # column name case is inconsistent - picking a convention and sticking with it
+    sheet_cols_to_lower(.) %>%
+
+    # a subset of years have extra sheets that are not needed - it was quicker to look them up and kick them out
+    remove_extra_sheets(., remove_sheets) %>%
+
+    # Column names are inconsistent bot between workbooks and within workbooks.
+    # A little column validation happens here
+    validate_col_names(.) %>%
+
+    # Because of the data format, many extra columns come along
+    # for the ride. This step limits us to our columns of interest
+    select_cols_to_keep(., keep_cols) %>%
+
+    # All datasets have a mix of data and column headers which causes all
+    # data to import as `chr`
+    coerce_numeric_columns(.) %>%
+
+    # Site IDs are not not backfilled in most of the data sets
+    backfill_site_ids(.) %>%
+
+    # Removes all partially populated row created by the
+    # previous two munge steps
+    remove_incomplete_rows(.) %>%
+    convert_date(.) %>%
+    dplyr::bind_rows(.)
+
+}
+
+
+# Helper functions -----------------------------------
+
+#' Read data in from all sheets in an excel file
+#'
+#' This function includes a bit of error handling to manage the following cases:
+#' (1) If the excel workbook contains empty sheets
+#' (2) If some of the files have problematic data in the first row and need to be skipped
+#'
+#' @param filepath chr, full file path
+#' @param subset_sheets chr vector, a subset of sheets that may need to skip a few rows to read properly
+#' @param subset_skip num, the number of rows that need to be skipped in order to read the files listed in `subset_sheets`
+#'
+read_usace_sheets <- function(filepath, subset_sheets = NULL, subset_skip = NULL) {
+
+  # read in data
+  dat <- filepath %>%
+    readxl::excel_sheets() %>% # get all sheet names for `read_xlsx`
+    purrr::set_names() %>% # name the data.frames in the list - skip?
+    purrr::map(read_xlsx, path = filepath, col_names = T)
+
+  # this functionality it required to read in all 2010 datasets
+  if(!is.null(subset_sheets)) {
+
+    # sanity check
+    if(is.null(subset_skip)){stop("`subset_skip` must be specified if `subset_sheets` is specified")}
+
+    # remove `subset_sheets` from the data set
+    dat <- dat[!(names(dat) %in% subset_sheets)]
+
+    # read in the sheets that have to skip `subset_skip` rows before reading
+    dat_subset <- filepath %>%
```

Comment:
The two weirdos include pieces of descriptive information in the first row instead of the data we are looking for.  Here is a screenshot of `PT` from `Missouri_USACE_2010_profiles.xlsx`:
![image](https://user-images.githubusercontent.com/15007294/156399715-90999ac4-5c02-4d33-a246-b670627dd09a.png)

This causes `read_xlsx` to start reading in the first row instead of skipping it like it would for a "normal" sheet (screenshot of `HD` from the 2010 dataset):
![image](https://user-images.githubusercontent.com/15007294/156401838-4266a0de-08d6-4508-b2e6-f8f164bdaa01.png)

I ran a modified version of the code you proposed above and compared the two data sets using `all.equal` to verify that the results of the two approaches match. Here is the updated snippet:
```r
dat_subset <- subset_sheets %>%
   purrr::set_names() %>% 
   purrr::map(readxl::read_xlsx, path = filepath, skip = subset_skip, col_names = T) # removed "set" and "subset_sheets"
```

I am going to update the function with your piece of code.